client/test/webview/SampleView.test.ts

```diff
@@ -0,0 +1,734 @@
+            mySandBox.stub(SampleView, 'getSample').returns(sample);
+            mySandBox.stub(SampleView, 'getContract').returns(sample.category.contracts[0]);
+            const pathExistsStub: sinon.SinonStub = mySandBox.stub(fs, 'pathExists').resolves(false);
+
+            await SampleView.openFile(repositoryName, 'FabCar', 'contracts', 'FabCar Contract', 'TypeScript');
+
+            repositoryRegistryGetStub.should.have.been.calledOnceWithExactly(repositoryName);
+            pathExistsStub.should.have.been.calledOnceWithExactly('/some/path');
+            repositoryRegistryDeleteStub.should.have.been.calledOnceWithExactly(repositoryName);
+            showErrorMessageStub.should.have.been.calledOnceWithExactly(`The location of the file(s) you're trying to open is unknown. The sample repository has either been deleted or moved. Try re-cloning the sample repository.`);
+        });
+
+        it('should return if user doesnt select how to open files', async () => {
+            const repositoryRegistryGetStub: sinon.SinonStub = mySandBox.stub(RepositoryRegistry.prototype, 'get').returns({name: repositoryName, path: '/some/path'});
+
+            mySandBox.stub(SampleView, 'getRepository').resolves({
+                name: 'hyperledger/fabric-samples',
+                remote: 'https://github.com/hyperledger/fabric-samples.git',
+                samples: [
+                    sample
+                ]
+            });
+            mySandBox.stub(SampleView, 'getSample').returns(sample);
+            mySandBox.stub(SampleView, 'getContract').returns(sample.category.contracts[0]);
+            const pathExistsStub: sinon.SinonStub = mySandBox.stub(fs, 'pathExists').resolves(true);
+            const shellCdStub: sinon.SinonStub = mySandBox.stub(shell, 'cd').returns(undefined);
+            const sendCommandStub: sinon.SinonStub = mySandBox.stub(CommandUtil, 'sendCommand').resolves();
+            mySandBox.stub(UserInputUtil, 'delayWorkaround').resolves();
+            mySandBox.stub(UserInputUtil, 'showFolderOptions').resolves();
+            const openNewProjectStub: sinon.SinonStub = mySandBox.stub(open, 'openNewProject').resolves();
+
+            await SampleView.openFile(repositoryName, 'FabCar', 'contracts', 'FabCar Contract', 'TypeScript');
+
+            repositoryRegistryGetStub.should.have.been.calledOnceWithExactly(repositoryName);
+            pathExistsStub.should.have.been.calledOnceWithExactly('/some/path');
+            shellCdStub.should.have.been.calledOnceWithExactly('/some/path');
+            sendCommandStub.should.have.been.calledOnceWithExactly('git checkout -b master origin/master');
+            openNewProjectStub.should.have.not.have.been.called;
+        });
+
+        it('should handle if the local branch already exists', async () => {
+            const repositoryRegistryGetStub: sinon.SinonStub = mySandBox.stub(RepositoryRegistry.prototype, 'get').returns({name: repositoryName, path: '/some/path'});
```

Comment:
could this be put in the beforeEach so you don't have it in each test, plus any other duplicate code